src/EntryPoint/RestEntryPoints.ts

```diff
@@ -52,29 +52,27 @@ export class RestEntryPoints implements EntryPoints {
     });
   }
 
-  public async entrypoints(
-    projectId: string,
-    limit: number = 10,
-    connectivity?: string[],
-    status?: string[]
-  ): Promise<EntryPoint[]> {
-    let remaining = limit;
+  public async entrypoints(filter: EntryPointFilter): Promise<EntryPoint[]> {
+    let remaining = filter.limit ?? 10;
     const data: EntryPoint[] = [];
     let nextId: string;
     let nextCreatedAt: string;
 
     while (remaining > 0) {
       const {
         data: { items = [] }
-      } = await this.client.get(`/api/v2/projects/${projectId}/entry-points`, {
-        params: {
-          nextId,
-          nextCreatedAt,
-          limit: Math.min(remaining, this.entrypointsPaginationBatchSize),
-          connectivity,
-          status
+      } = await this.client.get(
+        `/api/v2/projects/${filter.projectId}/entry-points`,
+        {
+          params: {
+            nextId,
+            nextCreatedAt,
+            connectivity: filter.connectivity,
+            status: filter.status,
+            limit: Math.min(remaining, this.entrypointsPaginationBatchSize)
+          }
```

Comment:
The same applies to the `params`:
```suggestion
          params: {
            nextId,
            nextCreatedAt,
            ...filters,
            limit: Math.min(remaining, this.entrypointsPaginationBatchSize)
          }
```

Reply:
oh, cool! didn't know that it's possible to do like this o_O